### PR TITLE
Implement IRLS

### PR DESCRIPTION
**Pull request by infrontoftheforest**

### CODE REVIEW

File: homework1.2/code/energy_function.m [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+% bisquare_func = @(e, c) (c^2)/6 - (e<=c).*((c^2)/6).*(1-(e./c).^2).^3;
+
+% Project 3D matches to image i+1 using current guess
+proj = worldToImage(A, R, t, M);
+% Compute the reprojection error
+dist = proj - m;
+e = [dist(:, 1)'; dist(:, 2)'];
+% interleave the vectors [u1; v1; u2; v2; ...]:
+e = e(:);
+sigma = 1.48257968*median(abs(e));
```
Comment (rifqifatih) [inline homework1.2/code/energy_function.m]: Is there a difference on using `median(abs(e))` and `mad(e)`? https://de.mathworks.com/help/stats/mad.html

File: homework1.2/code/energy_function.m [hunk 2]
```diff
@@ -0,0 +1,21 @@
+function [energy, W, e] = energy_function(A, R, t, M, m)
+
+bisquare_weights = @(e, c) (abs(e)<c).*(1 - e.^2/(c^2)).^2;
+bisquare_func = @(e, c) (c^2)/6 - (abs(e)<=c).*((c^2)/6).*(1-(e./c).^2).^3;
```
Comment (rifqifatih) [inline homework1.2/code/energy_function.m]: Any reference where the condition should be `abs(e)` instead of just `e`?

Comment (rifqifatih) [reply homework1.2/code/energy_function.m]: Well the error shouldn't be negative anyway..

File: homework1.2/code/energy_function.m (hunk 1 above)
Comment (infrontoftheforest) [reply homework1.2/code/energy_function.m]: I didn't realize that was a function.

File: homework1.2/code/energy_function.m (hunk 2 above)
Comment (infrontoftheforest) [reply homework1.2/code/energy_function.m]: I wrote a question on Moodle about it. I think the equation in the homework is wrong.

File: homework1.2/code/hw1_task3_skeleton.m [hunk 3]
```diff
@@ -102,33 +129,135 @@
...
+            M = xcoor(idx, :);
+            model.coord3d(points_found, :) = M(1, :);
+            model.descriptors(:, points_found) = descriptors{i}(:, j);
+        end
+    end
+    model.coord3d = model.coord3d(1:points_found, :);
+    model.descriptors = model.descriptors(:, 1:points_found);
+
+    % 2) Find SIFT matches
+    img = single(rgb2gray(imread(Filenames{i+1})));
```
Comment (rifqifatih) [inline homework1.2/code/hw1_task3_skeleton.m]: Any usage of the `img`?

**Pull request merged**